editor/src/components/canvas/controls/guideline-controls.tsx

```diff
@@ -153,3 +201,129 @@ function useGuideline<T = HTMLDivElement>(
   }, [innerCallback])
   return controlRef
 }
+
+function framesFromMetadata(
+  metadata: ElementInstanceMetadataMap,
+  paths: Array<ElementPath>,
+): Array<CanvasRectangle> {
+  return mapDropNulls((path) => metadata[EP.toString(path)]?.globalFrame ?? null, paths)
+}
+
+interface CanvasSpan {
+  a: CanvasPoint
+  b: CanvasPoint
+}
+
+function canvasSpan(a: CanvasPoint, b: CanvasPoint): CanvasSpan {
+  return { a, b }
+}
+
+function guidelineToSpan(guideline: Guideline): CanvasSpan | null {
+  switch (guideline.type) {
+    case 'XAxisGuideline': {
+      const a = canvasPoint({ x: guideline.x, y: guideline.yBottom })
+      const b = canvasPoint({ x: guideline.x, y: guideline.yTop })
+      return canvasSpan(a, b)
+    }
+    case 'YAxisGuideline': {
+      const a = canvasPoint({ x: guideline.xLeft, y: guideline.y })
+      const b = canvasPoint({ x: guideline.xRight, y: guideline.y })
+      return canvasSpan(a, b)
+    }
+    case 'CornerGuideline':
+      return null
+    default:
+      const _: never = guideline
+      throw new Error(`Unknown guideline found`)
```

Comment:
```suggestion
      throw new Error(`Unknown guideline of type ${guideline.type} found`)
```
Just so that we have a bit of extra info if this ever does happen

Reply:
what you suggested was my first idea too

Reply:
Aha of course! Ok this is fine then.